[BUG] Docker image build fails — TypeScript errors in admin-ui block image creation

## Description
Running `docker build -t authme .` fails at the admin-ui build step with TypeScript errors:

```
src/pages/auth-flows/AuthFlowEditorPage.tsx(52,9): error TS2322:
  Type 'string | null | undefined' is not assignable to type 'string | undefined'.

src/api/__tests__/authFlows.test.ts(15,10): error TS6133: 'makeFlow' is declared but never read.
src/components/__tests__/ConfirmDialog.test.tsx(2,26): error TS6133: 'fireEvent' is declared but never read.
src/components/__tests__/PasswordInput.test.tsx(2,26): error TS6133: 'fireEvent' is declared but never read.
src/test/utils.tsx(3,44): error TS6133: 'MemoryRouterProps' is declared but never read.
```

**The Docker image cannot be built at all.** This blocks all containerized deployments (Docker, Kubernetes, Helm).

## Root Causes
1. `AuthFlowEditorPage.tsx:52` — type mismatch: `null` not assignable to `string | undefined`
2. Test files are included in `tsc` build due to `noUnusedLocals` being enabled and test files not being excluded from compilation
3. Unused imports in test files cause TS6133 errors

## Steps to Reproduce
```bash
docker build -t authme-test .
```

## Affected Code
- `admin-ui/src/pages/auth-flows/AuthFlowEditorPage.tsx:52`
- `admin-ui/tsconfig.json` — includes test files in compilation

--- a/admin-ui/src/api/__tests__/authFlows.test.ts
+++ b/admin-ui/src/api/__tests__/authFlows.test.ts
@@ -7,34 +7,7 @@ import {
   createAuthFlow,
   updateAuthFlow,
   deleteAuthFlow,
-  type AuthFlow,
 } from '../authFlows';
-
-// ─── Helpers ─────────────────────────────────────────────────
-
-function makeFlow(overrides: Partial<AuthFlow> = {}): AuthFlow {
-  return {
-    id: 'flow-1',
-    realmId: 'realm-1',
-    name: 'Basic Login',
-    description: 'Password only',
-    isDefault: false,
-    steps: [
-      {
-        id: 'password-1',
-        type: 'password',
-        required: true,
-        order: 1,
-        condition: null,
-        fallbackStepId: null,
-        config: {},
-      },
-    ],
-    createdAt: '2024-01-01T00:00:00.000Z',
-    updatedAt: '2024-01-01T00:00:00.000Z',
-    ...overrides,
-  };
-}
 
 const BASE = '/admin';
 

--- a/admin-ui/src/components/__tests__/ConfirmDialog.test.tsx
+++ b/admin-ui/src/components/__tests__/ConfirmDialog.test.tsx
@@ -1,5 +1,5 @@
 import { describe, it, expect, vi } from 'vitest';
-import { render, screen, fireEvent } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 import ConfirmDialog from '../ConfirmDialog';
 

--- a/admin-ui/src/components/__tests__/PasswordInput.test.tsx
+++ b/admin-ui/src/components/__tests__/PasswordInput.test.tsx
@@ -1,5 +1,5 @@
 import { describe, it, expect } from 'vitest';
-import { render, screen, fireEvent } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 import PasswordInput from '../PasswordInput';
 

--- a/admin-ui/src/pages/auth-flows/AuthFlowEditorPage.tsx
+++ b/admin-ui/src/pages/auth-flows/AuthFlowEditorPage.tsx
@@ -49,7 +49,7 @@ export default function AuthFlowEditorPage() {
     mutationFn: (flow: Partial<AuthFlow>) =>
       updateAuthFlow(name!, flowId!, {
         name: flow.name,
-        description: flow.description,
+        description: flow.description ?? undefined,
         isDefault: flow.isDefault,
         steps: flow.steps,
       }),

--- a/admin-ui/src/test/utils.tsx
+++ b/admin-ui/src/test/utils.tsx
@@ -1,6 +1,6 @@
 import React from 'react';
 import { render, type RenderOptions } from '@testing-library/react';
-import { MemoryRouter, Routes, Route, type MemoryRouterProps } from 'react-router-dom';
+import { MemoryRouter, Routes, Route } from 'react-router-dom';
 import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
 
 interface WrapperOptions {

--- a/admin-ui/tsconfig.app.json
+++ b/admin-ui/tsconfig.app.json
@@ -24,5 +24,13 @@
     "noFallthroughCasesInSwitch": true,
     "noUncheckedSideEffectImports": true
   },
-  "include": ["src"]
+  "include": ["src"],
+  "exclude": [
+    "src/**/__tests__/**",
+    "src/**/*.test.ts",
+    "src/**/*.test.tsx",
+    "src/**/*.spec.ts",
+    "src/**/*.spec.tsx",
+    "src/test/**"
+  ]
 }

